quartz/util/imageHelper.ts

```diff
@@ -0,0 +1,115 @@
+import satori, { FontWeight, SatoriOptions } from "satori/wasm"
+import { GlobalConfiguration } from "../cfg"
+import { JSXInternal } from "preact/src/jsx"
+
+/**
+ * Get an array of `FontOptions` (for satori) given google font names
+ * @param headerFontName name of google font used for header
+ * @param bodyFontName name of google font used for body
+ * @returns FontOptions for header and body
+ */
+export async function getSatoriFont(headerFontName: string, bodyFontName: string) {
+  const headerWeight = 700 as FontWeight
+  const bodyWeight = 400 as FontWeight
+
+  // Fetch fonts
+  const headerFont = await fetchTtf(headerFontName, headerWeight)
+  const bodyFont = await fetchTtf(bodyFontName, bodyWeight)
+
+  // Convert fonts to satori font format and return
+  const fonts: SatoriOptions["fonts"] = [
+    { name: headerFontName, data: headerFont, weight: headerWeight, style: "normal" },
+    { name: bodyFontName, data: bodyFont, weight: bodyWeight, style: "normal" },
+  ]
+  return fonts
+}
+
+/**
+ * Get the `.ttf` file of a google font
+ * @param fontName name of google font
+ * @param weight what font weight to fetch font
+ * @returns `.ttf` file of google font
+ */
+function fetchTtf(fontName: string, weight: FontWeight): Promise<ArrayBuffer> {
+  return new Promise(async (resolve, reject) => {
+    // Get css file from google fonts
+    const css = await (
+      await fetch(`https://fonts.googleapis.com/css?family=${fontName}:${weight}`)
+    ).text()
+
+    // Extract .ttf url from css file
+    const urlRegex = /url\((https:\/\/fonts.gstatic.com\/s\/.*?.ttf)\)/g
+    const match = urlRegex.exec(css)
+    if (match) {
+      // fontData is an ArrayBuffer containing the .ttf file data
+      const fontData = await (await fetch(match[1])).arrayBuffer()
```

Comment:
this seems sort of brittle, why `match[1]`?

Reply:
A response from from google fonts looks something like this:
```ts
[
  'url(https://fonts.gstatic.com/s/schibstedgrotesk/v3/JqzK5SSPQuCQF3t8uOwiUL-taUTtarVKQ9vZ6pJJWlMNxcYwSMhq.ttf)',
  'https://fonts.gstatic.com/s/schibstedgrotesk/v3/JqzK5SSPQuCQF3t8uOwiUL-taUTtarVKQ9vZ6pJJWlMNxcYwSMhq.ttf',
  index: 98,
  input: '@font-face {\n' +
    "  font-family: 'Schibsted Grotesk';\n" +
    '  font-style: normal;\n' +
    '  font-weight: 700;\n' +
    "  src: url(https://fonts.gstatic.com/s/schibstedgrotesk/v3/JqzK5SSPQuCQF3t8uOwiUL-taUTtarVKQ9vZ6pJJWlMNxcYwSMhq.ttf) format('truetype');\n" +
    '}\n',
  groups: undefined
]
```

The url for the font is always contained twice, but only the second entry contains the "raw" version that doesn't require any additional string manipulation to retrieve it